packages/world/src/systems/MineSystem.sol

```diff
@@ -198,7 +198,29 @@ contract MineSystem is System {
     }
   }
 
-  function _destroyEntity(EntityId caller, EntityId mined, ObjectType minedType, Vec3 baseCoord) internal {
+  function _handleAbove(EntityId caller, Vec3 coord) internal {
+    // Remove growables on top of this block
+    Vec3 aboveCoord = coord + vec3(0, 1, 0);
+
+    (EntityId above, ObjectType aboveType) = EntityUtils.getOrCreateBlockAt(aboveCoord);
+
+    if (aboveType.isGrowable() && SeedGrowth._getFullyGrownAt(above) <= block.timestamp) {
+      // If the seed is fully grown, grow it and don't remove it yet
+      aboveType = NatureLib.growSeed(aboveCoord, above, aboveType);
+    }
+
+    if (aboveType.isLandbound()) {
+      _removeBlock(above, aboveType, aboveCoord);
+      _handleDrop(caller, above, aboveType, aboveCoord);
+    }
+  }
+
+  function _removeGrowable(EntityId entityId, ObjectType objectType, Vec3 coord) internal {
+    EntityObjectType._set(entityId, ObjectTypes.Air);
```

Comment:
why was the check for cannot mine fully grown seed removed here?

Reply:
because now we are always dealing with a non fully grown growable here:
- If the object being mined is growable and is fully grown, we grow it before removing
- If the object above being mined and is fully grown, we also grow it before removing

Reply:
gothca